src/controllers/controllermanager.cpp

```diff
@@ -168,13 +172,21 @@ void ControllerManager::updateControllerList() {
     locker.unlock();
 
     QList<Controller*> newDeviceList;
-    foreach (ControllerEnumerator* pEnumerator, enumerators) {
+    for (ControllerEnumerator* pEnumerator: enumerators) {
         newDeviceList.append(pEnumerator->queryDevices());
     }
 
+    // Since there is only one KeyboardController, it is not
+    // enumerated by an enumerator but added directly here.
+    KeyboardController* kbdController = new KeyboardController(m_pKeyboard);
+    newDeviceList.prepend(kbdController);
```

Comment:
Are you sure, this code is not called twice and we end up with two KeyboardController?